pkg/flow/logging/sink.go

```diff
@@ -95,5 +103,33 @@ func writerSinkLogger(w io.Writer, o SinkOptions) (log.Logger, error) {
 	if o.IncludeTimestamps {
 		l = log.With(l, "ts", log.DefaultTimestampUTC)
 	}
+
 	return l, nil
 }
+
+type fanoutWriter struct {
+	w io.Writer
+	f []loki.LogsReceiver
+}
+
+func (fw *fanoutWriter) Write(p []byte) (int, error) {
+	n, err := fw.w.Write(p)
+	if err != nil {
+		return n, err
+	}
+
+	for _, receiver := range fw.f {
+		select {
+		case receiver <- loki.Entry{
+			Labels: model.LabelSet{"component": "agent"},
+			Entry: logproto.Entry{
+				Timestamp: time.Now(),
+				Line:      string(p),
+			},
+		}:
+		default:
```

Comment:
This is to guard against issues like #2216, i.e. trying to send to the receiver of a component that is shutting down and getting GC'ed and its export is nil.

I'm not 100% if the logging block works similarly to components, will check one more time to make sure whether we need this.